### PR TITLE
LPS-178841 | Manage system object relationships endpoints when the related custom object is deactivated

### PR DESCRIPTION
**Pull request by Hijky**
**Background**:
Add a testcase to manage system object relationships endpoints when the related custom object is deactivated

**Test Plan**
CannotGetOneToManyRelationshipDetailsWhenCustomObjectIsDeactivated
Given a secondFoundation object
And Given a oneToMany relationship userSecondFoundations between a user object and a foundation object
When I deactivate the custom object
Then I receive 404 from the get request getUserAccount${relationshipName}Page
And Then the endpoint getUserAccount${relationshipName}Page is not visible in the OpenAPI

CannotGetManyToManyRelationshipDetailsWhenCustomObjectIsDeactivated
Given a foundation object
And Given a manyToMany relationship usersFoundations between a user object and a foundation object
When I deactivate the custom object
Then I can receive 404 from the get request getUserAccount${relationshipName}Page
And Then the endpoint getUserAccount${relationshipName}Page is not visible in the OpenAPI

CannotGetOneToManyRelatedEntryDetailsWhenCustomObjectIsDeactivated
Given a secondFoundation object
And Given a oneToMany relationship userSecondFoundations between a user object and a foundation object
And Given a user object entry
And Given two secondFoundation object entries
And Given secondFoundation entries related with userAccount through putUserAccountUserSecondFoundations
And Given foundations visible in getUserAccount${relationshipName}Page
When I deactivate secondFoundation object
Then the response of getUserAccount${relationshipName}Page is 404
And Then the endpoint getUserAccount${relationshipName}Page is not visible in the OpenAPI

CannotGetManyToManyRelatedEntryDetailsWhenCustomObjectIsDeactivated
Given a foundation object
And Given a manyToMany relationship usersFoundations between a user object and a foundation object
And Given two user object entries
And Given two foundation object entries
And Given  foundation entries related with userAccount through putUserAccountUsersFoundations
And Given  foundations visible in getUserAccount${relationshipName}Page
When I deactivate the custom object
Then the response of getUserAccount${relationshipName}Page is 404
And Then the endpoint getUserAccount${relationshipName}Page is not visible in the OpenAPI

**Jira ticket:**
https://issues.liferay.com/browse/LPS-172121

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

File: modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+				name = "userSecondFoundations",
+				objectDefinitionId1 = ${staticObjectId},
+				objectDefinitionId2 = ${secondFoundationObjectDefinitionId},
+				type = "oneToMany");
+		}
+
+		task ("Given manyToMany relationship userSecondFoundations created") {
+			var relationshipName = ObjectDefinitionAPI.createRelationship(
+				deletionType = "cascade",
+				en_US_label = "UsersFoundations",
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase:46]: userSecondFoundations in scenarios & UsersFoundations in codes

File: modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		for (var objectLabel : list "foundation,secondFoundation") {
+			ObjectAdmin.openObjectAdmin();
+
+			CreateObject.activeCustomObject(objectLabel = ${objectLabel});
+		}
+
+		ObjectAdmin.deleteAllCustomObjectsViaAPI();
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase:79]: Is deleting non-active custom object via API works?

Comment (Hijky) [reply modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase:79]: If the custom object is non-active, the deleted method doesn't work

File: modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+	@priority = 3
+	test CannotGetManyToManyRelationshipDetailsWhenCustomObjectIsDeactivated {
+		property portal.acceptance = "true";
+		property test.name.skip.portal.instance = "ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated#CannotGetManyToManyRelationshipDetailsWhenCustomObjectIsDeactivated";
+
+		task ("When I deactivate the custom object") {
+			ObjectAdmin.openObjectAdmin();
+
+			CreateObject.inactiveCustomObject(objectLabel = "foundation");
```
Comment (yanan) [inline modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase:142]: This is great to using UI as a workaround, do we have a ticket for inactive custom object doesn't work via API?

Comment (Hijky) [reply modules/apps/object/object-rest-test/src/testFunctional/tests/ManageSystemObjectRelationshipWhenRelatedCustomObjectIsDeactivated.testcase:142]: I think not, and I sent a question about "Can not set active=false by the manyToMany relationship via API" in the [slack channel](https://liferay.slack.com/archives/C026NQPV6TV/p1679312385674979)

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Review by magjed4289 — APPROVED** [review]

**Review by yanan — APPROVED** [review]

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**